docs/guides/end-to-end-testing/migration/selenium-to-cypress.mdx

```diff
@@ -0,0 +1,396 @@
+    cy.get('#loginButton').click()
+  })
+
+  it('Header is visible', () => {
+    cy.get('#header').should('be.visible')
+  })
+
+  it('Footer is visible', () => {
+    cy.get('#footer').should('be.visible')
+  })
+})
+```
+
+<br />
+Another big change is that you no longer need any WebDriver definitions. Cypress
+takes care of opening and closing the browser without any explicit prompting, unlike
+Selenium. Any code that is used to define the WebDriver is now obsolete.
+
+```java
+ WebDriver driver = new ChromeDriver();
+
+ WebDriverWait wait = new WebDriverWait(driver, 10);
+
+ driver.quit();
+```
+
+<br />
+Next would be to replace all the potential actions. The following code is fairly
+standard in Selenium frameworks.
+
+```python
+wait.until(EC.element_to_be_clickable((By.ID, "myButton"))).click()
+```
+
+<br />
+It is fairly easy to change this line of code to work inside a Cypress framework.
+It could even be done with a find and replace. In Cypress, the command would be the
+following:
+
+```Javascript
+cy.get('#myButton').click()
+```
+
+<br />
+Any wait until functionality that you have been using in Selenium is no longer required
+in Cypress. This functionality is [built in](/guides/core-concepts/retry-ability).
+
+The next vital part to migrate would be the [assertions](/guides/references/assertions). We can all agree assertions
+are moderately vital to a test automation framework. The next piece of code is a
+python Selenium assertion to validate if a button is visible. While it is possible
+to add third-party libraries, the built-in assertions of Cypress can tackle a wide
+range of situations.
+
+```python
+assert driver.find_element(By.ID, "myButton").is_displayed()
+```
+
+In Cypress, the same command would be:
+
+```javascript
+cy.get('#myButton').should('be.visible')
+```
+
+Considering the easy and clear syntax of Cypress, transferring all the selectors
+might take the longest. And while copying and pasting the selectors is definitely
+an option, rewriting the selectors might be required to conform to the best practices.
+
+##### iframe
+
+It's acknowledged that Cypress could handle iframes more effectively _out-of-the-box_. But, with the support of a robust
+community, solutions are readily available. In the case of iframes, the [cypress-iframe](https://www.npmjs.com/package/cypress-iframe)
+plugin was developed. This plugin will add a custom command to find and interact with an iframe.
+
+##### File upload
+
+Another community-based solution worth mentioning is file uploads. Although Cypress offers a built-in solution for
+handling [file uploads](/api/commands/selectfile), it may not be compatible with every upload component available.
+While `cy.get('input[type=file]').selectFile('file.json')` suffices in most scenarios, every so often, the web application requires a different
+approach. In such instances, plugins like this prove invaluable: [cypress-file-upload](https://www.npmjs.com/package/cypress-file-upload).
```

Comment:
[cypress-file-upload](https://www.npmjs.com/package/cypress-file-upload) is not listed on https://docs.cypress.io/plugins

It was last updated in June 2021 and is configured for legacy Cypress 3.0.0. The documentation is at legacy level.
Does it work with Cypress `13.x`?

Reply:
you're correct.  I found this blog which advises we replace the plugin with the in-built selectFile command.
https://www.cypress.io/blog/2022/01/19/uploading-files-with-selectfile

Reply:
Good find! Your blog article also links back to the documentation site https://docs.cypress.io/guides/references/migration-guide#Migrating-from-cypress-file-upload-to-selectFile as part of the section https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-100